Reconnect and device name fix for harmony platform

ehendrix23 opened the pull request:
## Description:

-) Increase aioharmony to 0.1.4 (fixes reconnect issue if connection lost)
-) Set PlatformNotReady if unable to connect to HUB on startup.
-) Fix when providing device name instead of id.

**Related issue (if applicable):** fixes #20010 

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.

If the code communicates with devices, web services, or third-party tools:
  - [x] New or updated dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.

ehendrix23 committed:
Update requirements

Updated requirements

ehendrix23 committed:
Add attributes

Add firmware and config version attributes

ehendrix23 committed:
Small bump for aioharmony

Small version bump increase for aioharmony

ehendrix23 committed:
Fix requirements file

For some reason aioharmony ended up in there as a duplicate. Fixed it.

ehendrix23 committed:
Fix for send command with named device

ehendrix23 committed:
Update requirements to get reconnect fix

ehendrix23 committed:
Set aioharmony version to 0.1.4

Version 0.1.4 has additional small fixes.

ehendrix23 committed:
Keep trying to connect on startup

Keep trying to connect to the HUB on startup

ehendrix23 committed:
Revert rebase changes

Revert some changes that should have been reverted back as part of rebase.

Kane610 commented on homeassistant/components/remote/harmony.py:
```diff
@@ -192,13 +192,19 @@ def __init__(self, name, host, port, activity, out_path, delay_secs):
+            try:
+                if await self._client.connect():
+                    break
+            except aioexc.TimeOut:
```
Wouldn't it be better to use PlatformNotReady?

ehendrix23 replied:
Two questions:
1. Is there an example you can point me to for PlatformNotReady?
2. If stating PlatformNotReady, will HASS then retry and what will it call then?

Thx. :-)

Kane610 replied:
Lots of examples in the code base.
PlatformNotReady will trigger setup of platform after a timer of 60 seconds I believe.

ehendrix23 committed:
PlatformNotReady if unable to connect on startup

Will call PlatformNotReady if unable to connect to Hub on startup

ehendrix23 committed:
Increase aioharmony requirement to 0.1.5

Increase aioharmony requirement to 0.1.5

MartinHjelmare commented on homeassistant/components/remote/harmony.py:
```diff
@@ -233,13 +247,15 @@ def new_activity(self, activity_info: tuple) -> None:
         self._state = bool(activity_id != -1)
         self._available = True
-        self.async_schedule_update_ha_state()
+        if self._platform_added:
```
Instead we should register the callbacks in `async_added_to_hass`. Can we register the callbacks after instantiating the client?

ehendrix23 replied:
It can be done afterwards. Do you want me to do that update in this PR then?

MartinHjelmare replied:
Great! Yes, please! Then we don't need to add `_platform_added`.

ehendrix23 replied:
LOL. Figured you would say that. Working on it right now. :-)

ehendrix23 replied:
Done, tested here to make absolutely sure every callback still works (verifying I did not have a bug in aioharmony <g>).

ehendrix23 committed:
Register callbacks when entity added to HASS

Register the callbacks only once the entity has been added to HASS instead of during setup of platform.

MartinHjelmare commented on homeassistant/components/remote/harmony.py:
```diff
@@ -106,6 +106,9 @@
+        if not await device.connect():
+            raise PlatformNotReady
+
         DEVICES.append(device)
```
A comment for a future PR:
Devices should not be stored in a constant but in hass.data.

MartinHjelmare commented on homeassistant/components/remote/harmony.py:
```diff
@@ -152,9 +155,7 @@ class HarmonyRemote(remote.RemoteDevice):
-        )
+        from aioharmony.harmonyapi import HarmonyAPI as HarmonyClient
 
         _LOGGER.debug("%s: Device init started", name)
```
We try to avoid side effects in init. We can remove this log. We log in `async_added_to_hass` too.

ehendrix23 replied:
Check failed due to another issue not related to my PR, hence to kick it off again had to do a new push. That _LOGGER line entry is removed. :-)

Build passed now, ready to be merged :-)

MartinHjelmare reviewed (approved):
Great! I'm noting some clean ups that would be good for the future.

MartinHjelmare commented:
Can be merged when build passes.

ehendrix23 committed:
Removed debug log in __init__

Removed debug log in __init__

maxalbani commented:
Hi, I'm new to home assistant.
Also I am experiencing the problem of frequent disconnections of the harmony component ..

How can I upgrade the component with this new release?
thanks
Max

MartinHjelmare commented:
Use the 0.86 beta release.
